File: week-02/day-2/src/IsInList.java
import java.util.*;

public class IsInList {

  public static void main(String... args) {
    ArrayList<Integer> list = new ArrayList<>(Arrays.asList(2, 4, 6, 8, 10, 12, 14, 16));
    System.out.println(checkNums(list));

    // Check if list contains all of the following elements: 4,8,12,16
    // Create a method that accepts list as an input
    // it should return "true" if it contains all, otherwise "false"

  }

  private static boolean checkNums(ArrayList<Integer> list) {
    // 4, 8, 12, 16     - all items to search
    ArrayList<Integer> nums = new ArrayList<>();
    nums.add(4);
    nums.add(8);
    nums.add(12);
    nums.add(16);

    if (list.containsAll(nums)) {
      return true;
    }
    return false;
  }
}

File: python_solutions/week-02/day-2/data_structures/is_in_list.py
# Check if list contains all of the following elements: 4,8,12,16
# Create a function that accepts list_of_numbers as an input
# it should return "True" if it contains all, otherwise "False"

list_of_numbers = [2, 4, 6, 8, 10, 12, 14, 16]


def check_nums(list_of_numbers: list):
    numbers_to_check = [4, 8, 12, 16]
    for number in numbers_to_check:
        if not list_of_numbers.__contains__(number):
            return False
    return True


if __name__ == '__main__':
    print(check_nums(list_of_numbers))
